src/databricks/labs/lsql/dashboards.py

```diff
@@ -819,30 +839,22 @@ def _format_query(query: str) -> str:
         return formatted_query
 
     @staticmethod
-    def create_dashboard(
-        dashboard_folder: Path,
-        *,
-        query_transformer: Callable[[sqlglot.Expression], sqlglot.Expression] | None = None,
-    ) -> Dashboard:
-        """Create a dashboard from code, i.e. configuration and queries.
+    def create_dashboard(dashboard_metadata: DashboardMetadata) -> Dashboard:
```

Comment:
it looks like the entire create_dashboard method from here should be converted to a method on `DashboardMetadata`:

```
def as_lakeview(self) -> Dashboard:
  self.validate()
  datasets = self.get_datasets() # make get_datasets private
  layouts = self.get_layouts() # make get_layouts private
  page = Page(
    name=dashboard_metadata.display_name,
    display_name=dashboard_metadata.display_name,
    layout=layouts,
  )
  return Dashboard(datasets=datasets, pages=[page])
```